From 07b3434f91c95d196f687c51e1a58a95a0b1ef2e Mon Sep 17 00:00:00 2001
From: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
Date: Wed, 28 May 2025 09:47:49 +0200
Subject: [PATCH] Don't add outlines to text when there is none

Qt SVG would indiscriminately add outlines to text. Since the pen
has Qt::NoBrush, the outline would not be visible with a normal
QPainter, but it did always trigger the QPainterPath code path which
is taken when the text uses features we don't support through the
native rasterizer backend.

In turn, this would cause all text to be unsearchable when printing
the SVG to a PDF. In addition, all text would be too thick in this
case because the PDF generator adds the outline even if the brush
is Qt::NoBrush.

Pick-to: 6.8
Fixes: QTBUG-123817
Change-Id: Icff637e65ea9f0c867a374fcbba3f6ad4a6756fc
Reviewed-by: Hatem ElKharashy <hatem.elkharashy@qt.io>
(cherry picked from commit 4d9bd9572f340fb9bc2f96f55ea6836386ac0b0d)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
(cherry picked from commit 09918e3da44f80bff8bb13cd65984bd7e00c95e6)
---

diff --git a/src/svg/qsvggraphics.cpp b/src/svg/qsvggraphics.cpp
index 67879da..7d6fca9 100644
--- a/src/svg/qsvggraphics.cpp
+++ b/src/svg/qsvggraphics.cpp
@@ -446,7 +446,8 @@
                 range.start = paragraphs.back().size();
                 range.length = newText.size();
                 range.format.setFont(font);
-                range.format.setTextOutline(p->pen());
+                if (p->pen().style() != Qt::NoPen && p->pen().brush() != Qt::NoBrush)
+                    range.format.setTextOutline(p->pen());
                 range.format.setForeground(p->brush());
 
                 if (appendSpace) {
